python/test/unit/language/test_core.py

```diff
@@ -844,6 +844,31 @@ def abs_kernel(X, Z, SIZE: tl.constexpr):
     torch.testing.assert_close(actual_f8, expect, equal_nan=True)
 
 
+# ----------------
+# test transpose
+# ----------------
+
+
+@pytest.mark.parametrize("dtype_x", [(dtype_x) for dtype_x in dtypes_with_bfloat16])
+def test_transpose(dtype_x, device='cuda'):
+    SIZE = 128
+
+    @triton.jit
+    def kernel(Z, X, SIZE: tl.constexpr):
+        off = tl.arange(0, SIZE)
+        off2d = off[None, :] + tl.arange(0, 1)[:, None]
+        x = tl.load(X + off2d)
+        z = x.T
+        tl.store(Z + off2d.T, z)
+
+    x = numpy_random([SIZE, 1], dtype_str=dtype_x)
```

Comment:
Might be better to try `[SIZE, 2]`? Since the transpose of `[SIZE, 1]` is just the same as the original tensor from the 1-d view.

Reply:
Done